include/caffe/data_layers.hpp

```diff
@@ -270,10 +270,13 @@ class MemoryDataLayer : public BaseDataLayer<Dtype> {
   virtual inline int ExactNumTopBlobs() const { return 2; }
 
```

Comment:
Define `Reshape` method to do the reshape of the tops, and use it in `DataLayerSetUp`. Remove the reshaping from the `Forward` since `Reshape` method is called by the net before calling `Forward`.